src/main/java/com/github/danielwegener/logback/kafka/delivery/AsynchronousDeliveryStrategy.java

```diff
@@ -16,16 +16,13 @@
     public <K, V, E> boolean send(Producer<K, V> producer, ProducerRecord<K, V> record, final E event,
                                   final FailedDeliveryCallback<E> failedDeliveryCallback) {
         try {
-            producer.send(record, new Callback() {
-                @Override
-                public void onCompletion(RecordMetadata metadata, Exception exception) {
-                    if (exception != null) {
-                        failedDeliveryCallback.onFailedDelivery(event, exception);
-                    }
+            producer.send(record, (metadata, exception) -> {
+                if (exception != null) {
+                    failedDeliveryCallback.onFailedDelivery(event, exception);
                 }
             });
             return true;
-        } catch (BufferExhaustedException | TimeoutException e) {
+        } catch (Exception e) {
```

Comment:
Since we can not predict in what context our logger is called  I am wondering if we then should handle the org.apache.kafka.common.errors.InterruptException as a special case and at least call `Thread.currentThread().interrupt()` such that a caller can react on the thread interruption.

```suggestion
        } catch (Exception e) {
          if (e instanceof org.apache.kafka.common.errors.InterruptException) {
            Thread.currentThread().interrupt();
          }
```

Reply:
I think that's fine. However I would rather forward the error and let the user choose what to do with it given that we don't know the caller's context like you said.